openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TestCaseRepository.java

```diff
@@ -233,6 +241,8 @@ public RestUtil.PutResponse<?> deleteTestCaseResult(String updatedBy, String fqn
       testCase.setTestCaseResult(storedTestCaseResult);
       ChangeDescription change = deleteTestCaseChangeDescription(testCase.getVersion(), storedTestCaseResult);
       ChangeEvent changeEvent = getChangeEvent(updatedBy, testCase, change, entityType, testCase.getVersion());
+      setTestSuiteSummary(testCase, timestamp, storedTestCaseResult.getTestCaseStatus(), true);
```

Comment:
@TeddyCr this code is bit confusing, why are we passing deleted as true?
are we looking at the testCase is deleted or not, if so shouldn't we check testCase.deleted instead of passing as another argument

Reply:
@harshach this is a testCaseResult -- not a testCase entity. If a testCase entity is deleted we handle it in the `postDelete` to remove its state from the testSuite.

As far as a testCaseResult goes it does not have a `deleted` field. We need to update the state of the testSuite entity and the testCase entity if we have a testCaseResult that is deleted and it is the latest one.

I think this is the downside of adding state to the entities. We have manage these cases.